packages/apps/reputation-oracle/server/src/modules/qualification/qualification.service.spec.ts

```diff
@@ -1,348 +1,337 @@
+          useValue: mockQualificationRepository,
+        },
+        {
+          provide: UserQualificationRepository,
+          useValue: mockUserQualificationRepository,
+        },
+        {
+          provide: UserRepository,
+          useValue: mockUserRepository,
         },
-        { provide: UserRepository, useValue: createMock<UserRepository>() },
         ServerConfigService,
+        ConfigService,
       ],
     }).compile();
 
-    qualificationService =
-      moduleRef.get<QualificationService>(QualificationService);
-    qualificationRepository = moduleRef.get<QualificationRepository>(
-      QualificationRepository,
-    );
-    userRepository = moduleRef.get<UserRepository>(UserRepository);
+    service = module.get<QualificationService>(QualificationService);
+  });
+
+  afterEach(() => {
+    jest.resetAllMocks();
   });
 
   describe('createQualification', () => {
-    it('should create a new qualification', async () => {
-      const createQualificationDto: CreateQualificationDto = {
-        reference: 'ref1',
-        title: 'title1',
-        description: 'desc1',
-        expiresAt: '2025-12-31T00:00:00.000Z',
+    it.each([faker.date.future({ years: 1 }), undefined])(
+      'should create a new qualification',
+      async (expiresAt) => {
+        const newQualification = {
+          title: faker.string.alpha(),
+          description: faker.string.alpha(),
+          expiresAt,
+        };
+
+        mockQualificationRepository.createUnique.mockResolvedValueOnce({
+          reference: faker.string.uuid(),
+          ...newQualification,
+        } as QualificationEntity);
+
+        await service.createQualification(newQualification);
+
+        expect(mockQualificationRepository.createUnique).toHaveBeenCalledTimes(
+          1,
+        );
+        expect(mockQualificationRepository.createUnique).toHaveBeenCalledWith(
+          expect.objectContaining({
+            ...newQualification,
+          }),
+        );
+      },
+    );
+
+    it('should throw a INVALID_EXPIRATION_TIME error', async () => {
+      const newQualification = {
+        title: faker.string.alpha(),
+        description: faker.string.alpha(),
+        expiresAt: faker.date.past(),
       };
 
-      const qualificationEntity = new QualificationEntity();
-      qualificationEntity.reference = createQualificationDto.reference;
-      qualificationEntity.title = createQualificationDto.title;
-      qualificationEntity.description = createQualificationDto.description;
-      qualificationEntity.expiresAt = new Date(
-        createQualificationDto.expiresAt!,
-      );
+      mockQualificationRepository.createUnique.mockResolvedValueOnce({
+        reference: faker.string.uuid(),
+        ...newQualification,
+      } as QualificationEntity);
+
+      try {
+        await service.createQualification(newQualification);
+      } catch (error) {
+        expect(error).toBeInstanceOf(QualificationError);
```

Comment:
`expect` should take place outside of `catch` block because if `createQualification` doesn't throw - they are not counted